Compilation fail macOS 13 (R 4.5)

Hello there!

I'm getting a build fail during [macOS 13 (devel)](https://github.com/albersonmiranda/fio/actions/runs/10538588481/job/29208097258) CI:

```
✖ Failed to build fs 1.6.4 (1.9s)
  Error: 
  ! error in pak subprocess
  Caused by error in `stop_task_build(state, worker)`:
  ! Failed to build source package fs.
  Full installation output:
  * installing *source* package ‘fs’ ...
  ** package ‘fs’ successfully unpacked and MD5 sums checked
  staged installation is only possible with locking
  ** using non-staged installation
  ** libs
  using C++ compiler: ‘Apple clang version 15.0.0 (clang-1500.1.0.2.5)’
  using SDK: ‘’
  clang++ -arch x86_64 -std=gnu++17 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./libuv-1.44.2/include -I. -pthread  -I/opt/R/x86_64/include    -fPIC  -falign-functions=64 -Wall -g -O2  -Wall -pedantic  -c dir.cc -o dir.o
  clang++ -arch x86_64 -std=gnu++17 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG -I./libuv-1.44.2/include -I. -pthread  -I/opt/R/x86_64/include    -fPIC  -falign-functions=64 -Wall -g -O2  -Wall -pedantic  -c error.cc -o error.o
  In file included from error.cc:1:
  In file included from ./error.h:5:
  In file included from /Library/Frameworks/R.framework/Resources/include/Rinternals.h:47:
  In file included from /Library/Frameworks/R.framework/Resources/include/R_ext/Error.h:30:
  In file included from /Library/Frameworks/R.framework/Resources/include/R_ext/RS.h:30:
  /Applications/Xcode_15.2.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/cstring:117:1: error: templates must have C++ linkage
  template <class _Tp>
  ^~~~~~~~~~~~~~~~~~~~
  /Library/Frameworks/R.framework/Resources/include/Rinternals.h:37:1: note: extern "C" language linkage specification begins here
  extern "C" {
  ^
  1 error generated.
  make: *** [error.o] Error 1
  ERROR: compilation failed for package ‘fs’
```
I've tried deleting the cache, but the error persisted. I'm not sure if it's on your side, but I thought it was worth reporting.

--- a/src/CollectorList.h
+++ b/src/CollectorList.h
@@ -1,3 +1,4 @@
+#include <R.h>
 #include <Rinternals.h>
 
 class CollectorList {

--- a/src/dir.cc
+++ b/src/dir.cc
@@ -9,6 +9,7 @@
 #undef ERROR
 
 #include "CollectorList.h"
+#include "R.h"
 #include "Rinternals.h"
 #include "error.h"
 #include "utils.h"

--- a/src/error.h
+++ b/src/error.h
@@ -2,6 +2,7 @@
 #define ERROR_H_
 
 #define R_NO_REMAP
+#include <R.h>
 #include <Rinternals.h>
 #undef R_NO_REMAP
 

--- a/src/file.cc
+++ b/src/file.cc
@@ -14,6 +14,7 @@
 #include <pwd.h>
 #endif
 
+#include <R.h>
 #include <Rinternals.h>
 
 #include "getmode.h"

--- a/src/fs.cc
+++ b/src/fs.cc
@@ -1,3 +1,4 @@
+#include "R.h"
 #include "Rinternals.h"
 #include "uv.h"
 

--- a/src/getmode.h
+++ b/src/getmode.h
@@ -1,6 +1,7 @@
 #pragma once
 
 #define R_NO_REMAP
+#include <R.h>
 #include <Rinternals.h>
 #undef R_NO_REMAP
 

--- a/src/id.cc
+++ b/src/id.cc
@@ -9,6 +9,7 @@
 #include <unistd.h>
 #endif
 
+#include <R.h>
 #include <Rinternals.h>
 
 #include "utils.h"

--- a/src/init.cc
+++ b/src/init.cc
@@ -1,6 +1,7 @@
 #include <stdlib.h> // for NULL
 #include <R.h>
 #include <R_ext/Rdynload.h>
+#include <R.h>
 #include <Rinternals.h>
 
 /* FIXME:

--- a/src/link.cc
+++ b/src/link.cc
@@ -1,6 +1,7 @@
 
 #include <cstring>
 
+#include "R.h"
 #include "Rinternals.h"
 #include "error.h"
 #include "utils.h"

--- a/src/path.cc
+++ b/src/path.cc
@@ -4,6 +4,7 @@
 #include <cstring>
 #include <string>
 
+#include "R.h"
 #include "Rinternals.h"
 #include "error.h"
 #include "utils.h"

--- a/src/unix/getmode.cc
+++ b/src/unix/getmode.cc
@@ -10,6 +10,7 @@
 #include "bsd/unistd.h" /* for getmode / setmode */
 #endif
 
+#include <R.h>
 #include <Rinternals.h> /* for Rf_error */
 
 #include <sys/stat.h>

--- a/src/windows/getmode.cc
+++ b/src/windows/getmode.cc
@@ -1,6 +1,7 @@
 #include "getmode.h"
 #include <sys/stat.h>
 
+#include <R.h>
 #include <Rinternals.h> /* for Rf_error */
 
 /* code adapted from https://cgit.freedesktop.org/libbsd/tree/src/setmode.c */

